### PR TITLE
Add validity checks to create

### PR DESCRIPTION
**Pull request by stevemessick**
@Hixie @eseidel 
https://github.com/flutter/flutter/issues/2802

Check that the given argument for project dir is not a file or non-empty directory or link.

### CODE REVIEW

**Comment by abarth** [comment]
@devoncarew

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -216,7 +220,30 @@ String _validateProjectName(String projectName) {
     return "Invalid project name: '$projectName' - this will conflict with Flutter "
     "package dependencies.";
   }
+  return null;
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: `return 0` here

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -216,7 +220,30 @@ String _validateProjectName(String projectName) {
...
+/// if we should disallow the directory name.
+String _validateProjectDir(String projectName) {
+  var type = FileSystemEntity.typeSync(projectName);
+  if (type != FileSystemEntityType.NOT_FOUND) {
+    switch(type) {
+      case FileSystemEntityType.DIRECTORY:
+        // Do not re-use directory if it is not empty.
+        if (new Directory(projectName).listSync(followLinks: false).isNotEmpty) {
+          return "Invalid project name: '$projectName' - refers to a directory "
+          "that is not empty.";
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: nit: two more spaces indent

File: packages/flutter_tools/test/create_test.dart [hunk 3]
```diff
@@ -48,5 +48,35 @@ void main() {
...
+      CreateCommand command = new CreateCommand();
+      CommandRunner runner = new CommandRunner('test_flutter', '')
+        ..addCommand(command);
+      var existingFile = new File("${temp.path.toString()}/bad");
+      if (!existingFile.existsSync()) existingFile.createSync();
+      await runner.run(['create', existingFile.path])
+          .then((int code) => expect(code, equals(1)));
+    },
+    // This test can take a while due to network requests.
+    timeout: new Timeout(new Duration(minutes: 2)));
```
Comment (devoncarew) [inline packages/flutter_tools/test/create_test.dart]: here and below, you can remove the timeout param - this is only necessary for the case where this (slow) test might succeed

**Comment by devoncarew** [comment]
Some nits, and some type annotations needed (https://travis-ci.org/flutter/flutter/builds/121765698#L326), then lgtm :)

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (stevemessick) [inline packages/flutter_tools/lib/src/commands/create.dart]: Pretty sure that's supposed to be null.

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 2 above)
Comment (stevemessick) [inline packages/flutter_tools/lib/src/commands/create.dart]: Not sure what you mean. The function? It is top-level. The extended string in the return? That's consistent with the other concatenated string in returns so I thought it should be that way.

File: packages/flutter_tools/test/create_test.dart (hunk 3 above)
Comment (stevemessick) [inline packages/flutter_tools/test/create_test.dart]: Done

**Comment by stevemessick** [comment]
Got rid of var's

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: Ah, yeah, I mixed this up w/ the `run()` method.

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 2 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: The continuation of the string on the next line; I think it needs as indent here - the existing string needs one as well :)

**Comment by stevemessick** [comment]
Added two spaces in two places

**Comment by devoncarew** [comment]
lgtm!

**Pull request merged**